dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/storage/datastore/DataStoreHandler.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.api.storage.datastore
+
+import com.datadog.android.core.internal.persistence.Deserializer
+import com.datadog.android.core.persistence.Serializer
+
+/**
+ * Interface for the datastore.
+ */
+interface DataStoreHandler {
+
+    /**
+     * Write data to the datastore.
+     *
+     * @param T datatype of the data to write to the datastore.
+     * @param key name of the datastore entry.
+     * @param data to write.
+     * @param version optional version for the entry.
+     * If not specified will give the entry version 0 - even if that would be a downgrade from the previous version.
+     * @param serializer to use to serialize the data.
+     */
+    fun <T : Any> setValue(
+        key: String,
+        data: T,
+        version: Int = 0,
+        serializer: Serializer<T>
+    )
+
+    /**
+     * Read data from the datastore.
+     *
+     * @param T datatype of the data to read from the datastore.
+     * @param key name of the datastore entry.
+     * @param version optional version to use when reading from the datastore.
+     * If specified, will only return data if the persistent entry exactly matches this version number.
+     * @param callback to return result asynchronously.
```

Comment:
worth mentioning that is will be called on the worker thread and not on the caller thread, maybe?